torch/fx/experimental/proxy_tensor.py

```diff
@@ -511,6 +513,26 @@ def set_original_aten_op(func):
 
 
 
+# This mode is **only** used for pre_dispatch tracing.
+# In particular, we need to make sure that autograd/autocast API's
+# that do not desugar into dispatcher operators stay in the graph.
+class PreDispatchTorchFunctionMode(TorchFunctionMode):
+
+    def __init__(self, tracer):
+        self.tracer = tracer
+
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs or {}
+        pre_dispatch_ops = [
+            torch._C._set_grad_enabled,
+            torch.amp._enter_autocast,
+            torch.amp._exit_autocast,
```

Comment:
Hello @bdhirsh is it accurate to say that by the time we create the output fx graph from dynamo (before running the compiler), all of the context managers / mutations to grad_enabled/autocast will have been canonicalized to these 3 functions?